src/jpgimage.cpp

```diff
@@ -48,12 +48,8 @@ namespace Exiv2 {
     {
         if (sizePsData < 4)
             return false;
-        for (auto&& i : irbId_) {
-            assert(strlen(i) == 4);
-            if (memcmp(pPsData, i, 4) == 0)
-                return true;
-        }
-        return false;
+        return std::all_of(irbId_.begin(), irbId_.end(),
```

Comment:
Should not this be `std::any_of` ? In the previous version we are returning `true` if _any_ of the elements in irbId_ matches.

Reply:
Yeah, my mistake.

Reply:
The funny thing is that it did not seem to affect the tests :fearful: , even though the tests seems to be cover that area of the code:
![image](https://user-images.githubusercontent.com/102645/156915059-b62dfccf-20cc-427b-9377-8e03b2532542.png)

Actually by looking at the coverage I noticed that we could just assume that the function is always called with `sizePsData == 4` and remove the first conditional. If we look at the usages of the function we have this:

```
luis@ryzenLinux:/media/linuxDev/programming/exiv2$ ag isIrb src/
src/jpgimage.cpp
47:    bool Photoshop::isIrb(const byte* pPsData, size_t sizePsData)
97:        while (position <= sizePsData - 12 && isIrb(pPsData + position, 4)) {

src/psdimage.cpp
190:            if (!Photoshop::isIrb(buf, 4))
441:            if (!Photoshop::isIrb(buf, 4))
```

Reply:
even funnier is that when converting from any_of to all_of, tests are now failing.

Reply:
hrm never mind. Seems it's just windows. I guess microsoft implements std::any_of differently...